File: Principles of Programming Language/HW12J/MilitaryPerson.java
public class MilitaryPerson {

    private String firstName;
    private String lastName;
    private String rank;
    private String serialNumber;
    private String branch;
    private boolean active;

    public MilitaryPerson()
    {
        firstName = "Matt";
        lastName = "McCullough";
        rank = "Corporate";
        serialNumber = "1234";
        branch = "The Greatest";
        active = false;
    }

    public MilitaryPerson(String f, String l, String r, String s, String b, boolean a)
    {
        firstName = f;
        lastName = l;
        rank = r;
        serialNumber = s;
        branch = b;
        active = a;
    }

    public String getBranch()
    {
        return branch;
    }
    public void setBranch(String b)
    {
        branch = b;
    }




    public String toString()
    {
        return "First Name: " + firstName + "\nLast Name: " + lastName + "\nRank: " + rank + "\nSerial Number: " + serialNumber + "\nBranch: " + branch + "\nActive: " + active;
    }




    public static void main(String[] args)
    {
        MilitaryPerson p1 = new MilitaryPerson("Dwight", "Eisenhower", "General", "DDE-123", "Army", false);
        MilitaryPerson p2 = new MilitaryPerson("John Paul", "Jones", "Admiral", "JPJ-234", "Navy", false);

        System.out.println(p1);
        System.out.println();
        System.out.println(p2);

    }

}

File: Principles of Programming Language/HW12P/MilitaryPerson.py
class MilitaryPerson:
    def __init__(self, firstName,lastName,rank,serialNumber,branch,active):
        self.firstName = firstName;
        self.lastName = lastName;
        self.rank = rank;
        self.serialNumber = serialNumber;
        self.branch = branch;
        self.active = active;


#This is the default constructor, but you cannot have mulitple constructors in Python. Sorry :(
    #def __init__(self):
    #    self.firstName = "Matt";
    #    self.lastName = "McCullough";
    #    self.rank = "Corporate";
    #    self.serialNumber = "1234";
    #    self.branch = "The Greatest";
    #    self.active = False;

    def get_branch(self):
        return self.branch;

    def set_branch(self, b):
        self.branch = b;

    def __del__(self):
        print("Military Person deleted");



p1 = MilitaryPerson("Gomer", "Pyle", "Private", "GMP-345", "Marines", True);
p2 = MilitaryPerson("Grace", "Hopper", "Rear Admiral", "GAH-712", "Navy", False);
print(p1.get_branch());
print(p2.get_branch());

